addOns/network/src/test/java/org/zaproxy/addon/network/ExtensionNetworkUnitTest.java

```diff
@@ -1089,6 +1092,76 @@ void shouldThrowIfCreatingHttpServerWithNullHandler() throws Exception {
+        // Given
+        extension.hook(mock(ExtensionHook.class));
+        HttpServerConfig config = null;
+        // When / Then
+        assertThrows(NullPointerException.class, () -> extension.createHttpServer(config));
+    }
+
+    @Test
+    void shouldCreateHttpServerWithZapApi() throws Exception {
+        // Given
+        AtomicBoolean requestReachedHandler = new AtomicBoolean();
+        HttpMessageHandler handler = (ctx, msg) -> requestReachedHandler.set(true);
+        extension.hook(mock(ExtensionHook.class));
+        HttpMessage msg = new HttpMessage(new HttpRequestHeader("GET http://zap/ HTTP/1.1"));
+        TestClient client =
+                new TextTestClient(
+                        "127.0.0.1",
+                        ch -> ch.pipeline().addFirst("http.client", new HttpClientCodec()));
+        HttpServerConfig config =
+                HttpServerConfig.builder()
+                        .setHttpMessageHandler(handler)
+                        .setServeZapApi(true)
+                        .build();
+        given(optionsParam.getApiParam()).willReturn(mock(OptionsParamApi.class));
+        // When
+        try (Server server = extension.createHttpServer(config)) {
+            int port = server.start(Server.ANY_PORT);
+            Channel channel = client.connect(port, null);
+            channel.writeAndFlush(msg).sync();
+            TextTestClient.waitForResponse(channel);
+            // Then
+            assertThat(requestReachedHandler.get(), is(equalTo(false)));
+        } finally {
+            client.close();
+        }
+    }
+
+    @Test
+    void shouldCreateHttpProxyWithZapApi() throws Exception {
+        // Given
+        AtomicBoolean requestReachedHandler = new AtomicBoolean();
+        HttpMessageHandler handler = (ctx, msg) -> requestReachedHandler.set(true);
+        extension.hook(mock(ExtensionHook.class));
+        HttpMessage msg = new HttpMessage(new HttpRequestHeader("GET http://zap/ HTTP/1.1"));
+        TestClient client =
+                new TextTestClient(
+                        "127.0.0.1",
+                        ch -> ch.pipeline().addFirst("http.client", new HttpClientCodec()));
+        HttpServerConfig config =
+                HttpServerConfig.builder()
+                        .setHttpMessageHandler(handler)
+                        .setHttpSender(new HttpSender(1))
+                        .setServeZapApi(true)
+                        .build();
+        given(optionsParam.getApiParam()).willReturn(mock(OptionsParamApi.class));
+        // When
+        try (Server server = extension.createHttpServer(config)) {
+            int port = server.start(Server.ANY_PORT);
+            Channel channel = client.connect(port, null);
+            channel.writeAndFlush(msg).sync();
+            TextTestClient.waitForResponse(channel);
+            // Then
+            assertThat(requestReachedHandler.get(), is(equalTo(false)));
```

Comment:
The request shouldn't reach the handler?
Same with the test above.

This seems confusing to me.

Reply:
Right, the API requests are handled by the ZAP API not the handler provided.

Reply:
Should there be a test that has a true result then?

Reply:
Oh I guess that's the test(s) down around 168?

Reply:
The existing tests (e.g. `shouldCreateHttpServerWithProvidedHandler`) check that the requests reach the handler provided (which don't have the ZAP API), the new tests check that the API is actually added.